src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py

```diff
@@ -74,22 +74,30 @@ class _Ssh(Lookup):
     def run(self, args, ssh_args, env_vars=None):
         if args.server == '-':
             args.server = 'django_manage[0]'
-        address = self.lookup_server_address(args)
-        if ':' in address:
-            address, port = address.split(':')
+        user, host, port = self.get_address_tuple(args)
+        address = f"{user}@{host}"
+        if port:
             ssh_args = ['-p', port] + ssh_args
-        if '@' in address:
-            username, address = address.split('@', 1)
-            username = get_ssh_username(address, args.env_name, requested_username=username)
-        elif '@' not in address:
-            username = get_ssh_username(address, args.env_name)
-        address = f"{username}@{address}"
         cmd_parts = [self.command, address, '-t'] + ssh_args
         cmd = ' '.join(shlex_quote(arg) for arg in cmd_parts)
         if not args.quiet:
             print_command(cmd)
         return subprocess.call(cmd_parts, **({'env': env_vars} if env_vars else {}))
 
+    def get_address_tuple(self, args):
+        address = self.lookup_server_address(args)
+        if ':' in address:
+            address, port = address.split(':')
```

Comment:
The scenario of `some.host:` will not raise an exception here and will return an an empty non-None port (i.e. `port == ''`). Is that ok (in the scope of this function)?